### PR TITLE
The Deep Bag of Holding

### PR DESCRIPTION
**Pull request by jbox144**
## About The Pull Request

Presenting the latest and greatest ~~fuckup~~ product of the science department, the Deep Bag of Holding! Made using [REDACTED] and a bag of holding, the Deep Bag of Holding opens up into a miniature pocket dimension, suitable for any needs!

The Deep Bag of Holding can be used on mobs, movable atoms and items to shove them into the pocket dimension. Mobs will incur a small delay so you can't go about shoving anyone and everyone in the bag. You can also shove yourself in the bag by using it on yourself. Items can be done instantly. 

After sending yourself to ~~brazil~~ your pocket dimension, you can construct whatever you want inside. The possibilities are endless! Drug labs! Mobile brig! Mobile medbay! Bagged SWAT team! Tritium burn chamber! Supermatter! Singulo! Tesla! ~~ERP den~~!

Destroying the bag will destroy your pocket dimension too! This will eject all items and mobs, but will not save machines and structures!

How do you get your hands on a deep bag of holding? Simple!

Get a regular bag of holding, and insert a bluespace anomaly core. Then, get a shocked grill with access to about 2MW of power (from testing at least), and shock yourself on it while you have the bag. The anomaly will then react with the bag and the high voltages, creating a deep bag of holding.

## Why It's Good For The Game

The Deep Bag of Holding is just a fun concept, and something for people to mess around with, and try to acquire once it's creation is implemented. It's also just a better version of the bluespace jar.

## Changelog
:cl:
add: Added the Deep Bag of Holding
add: Added "Deep Bluespace"
imageadd: Added the Deep Bag of Holding icons
/:cl:

### CODE REVIEW

**Comment by ZombiesVsPlants** [comment]
the shocking bag thing is pretty based and works towards the project

**Comment by jbox144** [comment]
> 
> 
> the shocking bag thing is pretty based and works towards the project

If I do that I might make it so it works if you shock a bag containing an anomally core, so you can't mass-produce them (well, easily)

**Comment by jupyterkat** [comment]
maxcap if you put another deep boh into it

**Comment by fryz1** [comment]
I have an idea and a question. What happens if a small explosion happens like a small potassium and water explosion? The idea is a way to make the deep boh maybe like a way of stabilizing the tear in the fabric of reality like if you put a bluespace anom in one of the boh and put a boh in it and that would make the deep boh instead of the tear.

**Comment by thelaughingbomb** [comment]
this sounds incredibly epic, also could we do recursive deep bags of holding?

**Comment by jbox144** [comment]
> 
> 
> this sounds incredibly epic, also could we do recursive deep bags of holding?

unsure. Probably not due to balance concerns, but eh.

**Comment by TheFakeElon** [comment]
> unsure. Probably not due to balance concerns

This entire PR is a balance concern.

**Comment by ZombiesVsPlants** [comment]
so do it anyway tbh

I want to see a demiplane filled with crates and cardboard boxes filled with mousetraps and in 1 of them is the nuke disk

**Comment by TheFakeElon** [comment]
Nuke disk can't leave the station z level

**Review by TheFakeElon — CHANGES_REQUESTED** [review]
Putting a deep bag in a deep bag should do something, preferably more devastating than just a normal tear. You also shouldn't be able to put these in normal bags of holding.

**Comment by Singularity69** [comment]
makes it so putting A DBoH into A DBoH will make A SM singulo?

**Comment by ZombiesVsPlants** [comment]
How do RCD's work on the floor? Will it turn into space?

**Comment by jbox144** [comment]
> 
> 
> How do RCD's work on the floor? Will it turn into space?

yes

**Comment by ZombiesVsPlants** [comment]
oh HELL yes

**Comment by TheFakeElon** [comment]
Is this still being worked on?

**Comment by jbox144** [comment]
yes

**Comment by jbox144** [comment]
Working on making it makeable as of right now. Should commit it soon-ish

File: austation/code/game/objects/items/deepbagofholding.dm [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+	message_admins("[ADMIN_LOOKUPFLW(user)] detonated a DEEP bag of holding at [ADMIN_VERBOSEJMP(loc)]. Failure state: [failure_state]")
+	log_game("[key_name(user)] detonated a DEEP bag of holding at [loc_name(loc)]. Failure state: [failure_state]")
+
+	switch (failure_state)
+		if (1) // Nothing happens
+			user.visible_message("<span class='warning'>[user] inserts [A.name] into [src], causing it to fizzle out of existence!</span>", "<span class='warning'>[src] fizzles out of existence! What a waste!</span>")
+		if (2) // The user is destroyed with the bags
+			user.visible_message("<span class='warning'>[user] inserts [A.name] into [src], causing them to fizzle out of existence!</span>", "<span class='warning'>You feel your body being dragged out of space and time!</span>")
+			qdel(user)
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/deepbagofholding.dm:160]: use dust(user)

File: austation/code/game/objects/items/deepbagofholding.dm [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+		if(safety != "Proceed" || QDELETED(A) || QDELETED(src) || QDELETED(user) || !user.canUseTopic(A, BE_CLOSE, iscarbon(user))) // need to be holding the bag you're "inserting"
+			return 1
+		var/turf/loccheck = get_turf(A)
+		if(is_reebe(loccheck.z))
+			user.visible_message("<span class='warning'>An unseen force knocks [user] to the ground!</span>", "<span class='big_brass'>\"I think not!\"</span>")
+			user.Paralyze(60)
+			return 1
+		failure(A, user)
+		qdel(A)
+		qdel(src)
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/deepbagofholding.dm:144]: should probably put these at the end of the failure proc

**Review by TheFakeElon — CHANGES_REQUESTED** [review]
Use booleans in return values instead of 1s and 0s

**Comment by jbox144** [comment]
> 
> 
> Use booleans in return values instead of 1s and 0s

done!

**Review by TheFakeElon — APPROVED** [review]
just remove the spaces between the `if` arguments and you're good to go

File: austation/code/game/objects/items/deepbagofholding.dm [hunk 3]
```diff
@@ -0,0 +1,239 @@
+//  Deep bag of holding. Still unsure as to how it will be made.
+
+/obj/item/deepbackpack
+	name = "deep bag of holding"
+	desc = "A bluespace bag, with a pocket dimension large enough to live in. It's huge!"
+	// Alternative descriptions: "Enslaved Tardis", "It's bigger on the inside", "Mobile ERP den", "Mobile dormitory", "Mini drug lab", e.t.c
```
Comment (MCterra10) [inline austation/code/game/objects/items/deepbagofholding.dm:6]: should have gone for mobile erp den in my honest opinion.

**Review by MCterra10 — APPROVED** [review]

File: austation/code/game/objects/items/deepbagofholding.dm (hunk 3 above)
Comment (jbox144) [reply austation/code/game/objects/items/deepbagofholding.dm:6]: Too direct. Plus, I figured that wouldn't get elon's vote.

Comment (TheFakeElon) [reply austation/code/game/objects/items/deepbagofholding.dm:6]: :clap: not :clap: an :clap: erp :clap: server :clap:

File: austation/code/game/objects/items/deepbagofholding.dm [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+	// Links the minidimension to the bag and vis-versa
+	for(var/turf/open/indestructible/deepBluespaceExit/exit in currentArea)
+		exit.parent = src
+	entry =	locate(linkedDim.bottom_left_coords[1] + linkedDimTemp.landingZoneRelativeX, \
+		    	  linkedDim.bottom_left_coords[2] + linkedDimTemp.landingZoneRelativeY, \
+		   		  linkedDim.bottom_left_coords[3])
+
+	// Start processing air and shit
+	START_PROCESSING(SSobj, src)
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/deepbagofholding.dm:39]: Is this redundant?

Comment (jbox144) [reply austation/code/game/objects/items/deepbagofholding.dm:39]: yes

Comment (TheFakeElon) [reply austation/code/game/objects/items/deepbagofholding.dm:39]: could you remove it then :P

Comment (jbox144) [reply austation/code/game/objects/items/deepbagofholding.dm]: Alright, done

**Pull request merged**